src/common/snippets/src/pass/lowered/load_movebroadcast_to_broadcastload.cpp

```diff
@@ -0,0 +1,63 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/itt.hpp"
+
+#include "snippets/pass/lowered/load_movebroadcast_to_broadcastload.hpp"
+#include "snippets/snippets_isa.hpp"
+
+namespace ngraph {
+namespace snippets {
+namespace pass {
+namespace lowered {
+
+
+LoadMoveBroadcastToBroadcastLoad::LoadMoveBroadcastToBroadcastLoad() {}
+
+bool LoadMoveBroadcastToBroadcastLoad::run(LoweredExprIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::LoadMoveBroadcastToBroadcastLoad")
+    bool modified = false;
+
+    for (auto expr_it = linear_ir.begin(); expr_it != linear_ir.end(); expr_it++) {
+        const auto &op = (*expr_it)->get_node();
+        // Match on MoveBroadcast because MoveBroadcast is rare node in bodies
+        if (const auto move_broadcast = ov::as_type_ptr<op::BroadcastMove>(op)) {
+            const auto interm_td = (*expr_it)->get_inputs().front();
+            const auto parent_expr = linear_ir.get_expr_by_output(interm_td).first;
+            const auto load = ov::as_type_ptr<op::Load>(parent_expr->get_node());
+            if (!load)
+                continue;
+
+            // Cannot rewrite Broadcast + Load if load has more than 1 user
+            // or more than one input, or if Broadcast has several inputs
+            const auto load_consumers_inputs = linear_ir.get_exprs_by_input(interm_td);
+            size_t count = 0;
+            for (const auto& consumer_expr_input : load_consumers_inputs) {
+                const auto consumer = consumer_expr_input.first->get_node();
+                if (ov::is_type<op::LoopEnd>(consumer))
+                    continue;
```

Comment:
Very minor: maybe a little more readable
```suggestion
                    if (!ov::is_type<op::LoopEnd>(consumer))
                          count++;
```

Reply:
Agree. Seems better. Done, thanks